packages/orchestration/src/exos/local-orchestration-account.js

```diff
@@ -265,32 +263,48 @@ export const prepareLocalOrchestrationAccountKit = (
        */
       returnVoidWatcher: {
         /**
-         * @param {Record<unknown, unknown>[]} results
+         * @param {any} _result
          */
-        onFulfilled(results) {
-          results.length === 1 ||
-            Fail`expected exactly one result; got ${results}`;
-          trace('Result', results[0]);
+        onFulfilled(_result) {
           return undefined;
         },
       },
+      /**
+       * handles a request for balance from a bank purse and returns the balance
+       * as a Chain Amount
+       */
+      getBalanceWatcher: {
+        /**
+         * @param {Amount<'nat'>} natAmount
+         * @param {{ denom: DenomAmount['denom'] }} ctx
+         * @returns {DenomAmount}
+         */
+        onFulfilled(natAmount, { denom }) {
+          return harden({ denom, value: natAmount.value });
+        },
+      },
       holder: {
-        /** @type {OrchestrationAccount<any>['getBalance']} */
-        async getBalance(denomArg) {
+        /**
+         * TODO: balance lookups for non-vbank assets
+         *
+         * @type {PromiseToVow<OrchestrationAccount<any>['getBalance']>}
+         */
+        getBalance(denomArg) {
           // FIXME look up real values
           // UNTIL https://github.com/Agoric/agoric-sdk/issues/9211
           const [brand, denom] =
             typeof denomArg === 'string'
               ? [/** @type {any} */ (null), denomArg]
               : [denomArg, 'FIXME'];
 
-          const natAmount = await E.when(
+          return watch(
             E(this.state.account).getBalance(brand),
+            this.facets.getBalanceWatcher,
+            { denom },
           );
-          return harden({ denom, value: natAmount.value });
         },
         getBalances() {
-          throw new Error('not yet implemented');
+          return asVow(() => Fail`not yet implemented`);
```

Comment:
Isn't there a helper already (as with `Promise.reject`) for making and already-rejected vow?

Reply:
Not to my knowledge. I think we could also do :
```js
watch(Promise.reject(Fail`...`))
```
But `asVow` seems concise enough to me here